### PR TITLE
Update milli (v0.3.1) with filterable attributes

### LINKED ISSUE #187: Rename attributesForFaceting into filterableAttributes
Related to the following spec:https://github.com/meilisearch/specifications/blob/main/text/0032-distinct-attribute.md

### LINKED ISSUE #195: Test reset settings
During the reset setting test:
- add a document
- after resetting, check the document is well returned (e.g. all the attributes are displayed)

### PR DESCRIPTION
**Pull request by curquiza**
Fixes #187 and #70
Also fixes #195

### CODE REVIEW

File: meilisearch-http/src/index/search.rs [hunk 1]
```diff
@@ -273,15 +273,15 @@ impl Matcher for HashSet<String> {
 
 impl Matcher for MatchingWords {
     fn matches(&self, w: &str) -> bool {
-        self.matches(w)
+        self.matching_bytes(w).is_some()
```
Comment (Kerollmops) [inline meilisearch-http/src/index/search.rs:276]: This must be handled by the highlighter at some point.

File: meilisearch-http/src/index/updates.rs [hunk 2]
```diff
@@ -51,7 +51,7 @@ pub struct Settings<T> {
         deserialize_with = "deserialize_some",
         skip_serializing_if = "Option::is_none"
     )]
-    pub attributes_for_faceting: Option<Option<HashMap<String, String>>>,
+    pub attributes_for_faceting: Option<Option<HashSet<String>>>,
```
Comment (Kerollmops) [inline meilisearch-http/src/index/updates.rs:54]:
```suggestion
    pub filterable_attributes: Option<Option<HashSet<String>>>,
```

File: meilisearch-http/src/routes/settings/mod.rs [hunk 3]
```diff
@@ -75,7 +75,7 @@ macro_rules! make_setting_route {
 
 make_setting_route!(
     "/indexes/{index_uid}/settings/attributes-for-faceting",
```
Comment (Kerollmops) [inline meilisearch-http/src/routes/settings/mod.rs:77]:
```suggestion
    "/indexes/{index_uid}/settings/filterable-attributes,
```

File: meilisearch-http/tests/settings/get_settings.rs [hunk 4]
```diff
@@ -19,7 +19,7 @@ async fn get_settings() {
     assert_eq!(settings.keys().len(), 6);
     assert_eq!(settings["displayedAttributes"], json!(["*"]));
     assert_eq!(settings["searchableAttributes"], json!(["*"]));
-    assert_eq!(settings["attributesForFaceting"], json!({}));
+    assert_eq!(settings["attributesForFaceting"], json!([]));
```
Comment (Kerollmops) [inline meilisearch-http/tests/settings/get_settings.rs:22]:
```suggestion
    assert_eq!(settings["FilterableAttributes"], json!([]));
```

**Comment by MarinPostma** [comment]
@curquiza can you rebase please?

**Comment by curquiza** [comment]
@MarinPostma done!

**Review by MarinPostma — APPROVED** [review]
thanks!

**Comment by MarinPostma** [comment]
bors merge

**Pull request merged**